File: app/src/main/java/com/apps/darkstorm/swrpg/sw/stuff/CriticalInjury.java
package com.apps.darkstorm.swrpg.sw.stuff;

import java.util.ArrayList;

public class CriticalInjury{
    //Version 1 0-2
    public String name;
    public String desc;
    public int severity;
    public Object serialObject(){
        ArrayList<Object> tmp = new ArrayList<>();
        tmp.add(name);
        tmp.add(desc);
        tmp.add(severity);
        return tmp.toArray();
    }
    public void loadFromObject(Object obj){
        Object[] tmp = (Object[])obj;
        switch (tmp.length){
            case 3:
                name = (String)tmp[0];
                desc = (String)tmp[1];
                severity = (int)tmp[2];
        }
    }
    public boolean equals(Object obj){
        if (obj instanceof CriticalInjury){
            CriticalInjury in = (CriticalInjury)obj;
            return in.name.equals(name) && in.desc.equals(desc) && in.severity == severity;
        }else{
            return false;
        }
    }
}

File: app/src/main/java/com/apps/darkstorm/swrpg/assistant/swElements/CriticalInjury.kt
package com.apps.darkstorm.swrpg.assistant.swElements

import android.util.JsonReader
import android.util.JsonToken
import android.util.JsonWriter
import com.apps.darkstorm.swrpg.assistant.saveLoad.JsonSavable

data class CriticalInjury(var name: String = "",var desc: String = "", var severity: Int = 0): JsonSavable(){
    override fun save(jw: JsonWriter) {
        jw.beginObject()
        jw.name("name").value(name)
        jw.name("description").value(desc)
        jw.name("severity").value(severity)
        jw.endObject()
    }
    override fun load(jr: JsonReader) {
        jr.beginObject()
        while(jr.hasNext() && jr.peek() != JsonToken.END_OBJECT){
            if(jr.peek()!= JsonToken.NAME){
                jr.skipValue()
                continue
            }
            val jName = jr.nextName()
            when(jName){
                "name"-> name = jr.nextString()
                "description"-> desc = jr.nextString()
                "severity"-> severity = jr.nextInt()
            }
        }
        jr.endObject()
    }
    override fun clone() = copy()
}